bundles/framework/layerselector2/Flyout.js

```diff
@@ -510,75 +471,127 @@ Oskari.clazz.define('Oskari.mapframework.bundle.layerselector2.Flyout',
+                        me.activateFilter(filterName);
+                        me._setFilterTooltip(filterName, loc.tooltips.remove);
+                    } else {
+                        // Deactivate all filters
+                        me.deactivateAllFilters();
+                    }
+                });
 
+                filterContainer.append(filterButton);
             });
+        },
 
-            me.filters.push(filter);
-            filterContainer.append(filterButton);
+        /**
+         * Set filter button tooltip
+         * @method  @private _setFilterTooltip
+         * @param {String} filterName filter name
+         * @param {String} tooltip    tooltip
+         */
+        _setFilterTooltip: function(filterName, tooltip) {
+            var me = this;
+            me.layerTabs.forEach(function(tab){
+                var filterContainer = tab.getTabPanel().getContainer().find('.layerselector2-layer-filter');
+                var filterIcon = filterContainer.find('.filter-icon.' + 'filter-'+filterName);
+                filterIcon.parents('.filter').attr('title',tooltip);
+            });
+        },
+        /**
+         * Set filter icon classes
+         * @method  @private _setFilterIconClasses
+         * @param {String} filterName filter name
+         */
+        _setFilterIconClasses: function(filterName) {
+            var me = this;
+            me.layerTabs.forEach(function(tab){
+                var filterContainer = tab.getTabPanel().getContainer().find('.layerselector2-layer-filter');
+                me.filters.forEach(function(filter){
+                    var filterIcon = filterContainer.find('.filter-icon.' + 'filter-'+filter.filterName);
+                    // First remove all active classes
+                    filterIcon.removeClass(filter.iconClassActive);
+                    filterIcon.removeClass(filter.iconClassDeactive);
+                    filterIcon.removeClass('active');
+                    // If filter has same than currently selected then activate icon
+                    if(filter.filterName === filterName) {
+                        filterIcon.addClass(filter.iconClassActive);
+                        filterIcon.addClass('active');
+                    }
+                    // Otherwise use deactive icon
+                    else {
+                        filterIcon.addClass(filter.iconClassDeactive);
+                    }
+                });
+            });
         },
+
+        /**
+         * Get currently selected filter.
+         * @method  @private _getCurrentFilter
+         * @param  {String} filterName filter name
+         * @return {Object|null}  filter object if founded or null
+         */
+        _getCurrentFilter: function(filterName) {
+            var me = this;
+            // Checks at if filter founded then return this
+            var currentFilter = jQuery.grep(me.filters || [], function(filter){
```

Comment:
Maybe use the native forEach instead of jQuery as the logic is pretty much the same?